libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java

```diff
@@ -370,6 +402,19 @@ public void addRemotePublisher(Class<? extends Transform> transformer,
         remotes.put(transformer, publisher);
     }
 
+    /**
+     * Removes a remote publisher to publish events to.
+     *
+     * @param transformer Transformer class.
+     */
+    void removeRemotePublisher(Class<? extends Transform> transformer) {
```

Comment:
Not being able to remove the remote publisher associated with a transformer is by design. You can either remove both the transformer and publisher, or keep them both. Could you remove this method? It won't be able to publish in your test scenario, since it's using an unauthenticated instance anyway.

Reply:
This method does remove both the transformer and the publisher, they are stored in a HashMap with the transformer acting as the key.

If I am not able to remove the default transformer + publisher I would need to build a full instrumentation payload which would make the tests much more verbose.

Reply:
@npars I'd still prefer to remove this method and not add it just for testing. I'm okay with verbose tests. There should be other tests already that construct a full instrumentation payload. Feel free to copy from there.

Reply:
I just tested with leaving `AILTNPublisher` active and it results in test failures since it tries to make an outbound network request, which rightly fails to succeed.

Reply:
@npars Yes, since we're testing publish, I'm suggesting that we make it a true live network test. We have examples of this in our test suite. All our `RestClient` tests are going against a live server and making calls. You can run these tests against your test org. Look at `RestClientTest` on how to setup a live network test and run it.

Reply:
The goal of this test isn't to validate the actual network request, but to validate batching of metrics prior to sending to the publisher. Making this an integration test as you suggest would add a lot of complexity and make the test more difficult to maintain.

Reply:
Alright, that's fair. I'll get this merged then.